### PR TITLE
Fix inconsistent padding behavior in Torch pooling operations

### PR DESCRIPTION
**Pull request by buildwithsuhana**
## Description

This PR resolves a discrepancy in the Torch backend's `average_pool` implementation. When utilizing asymmetric `"same"` padding, the Torch backend previously produced mathematically inconsistent and biased results compared to JAX, TensorFlow, and standard Keras behavior. 

To fix this, we implement the **Division Trick** (Zero padding followed by normalization by valid counts). This ensures localized averages are calculated accurately across boundary conditions without introducing edge bias.

---

#### Bug (Pre-Fix Behavior)
Previously, the Torch backend relied on naive average pooling, which suffered from edge artifacts when manual padding was required (e.g., `pool_size > 2` with asymmetric padding).

Consider an input $x = [10.0, 20.0]$ with `pool_size=3`, `strides=2`, and `padding="same"`. To satisfy the `"same"` output dimensions, one element of padding must be added on the right.

1. **Replication Padding (Biased Average):** If boundary elements are replicated to fulfill the padding size:
   $$\text{Window} = [10.0, 20.0, \mathbf{20.0}]$$
   $$\text{Average} = \frac{10 + 20 + 20}{3} = \mathbf{16.67}$$
   *(Output is heavily biased toward the edge value).*

2. **Naive Zero Padding (Bias towards Zero):** If filled with zeros but using a fixed window count divisor:
   $$\text{Window} = [10.0, 20.0, \mathbf{0.0}]$$
   $$\text{Average} = \frac{10 + 20 + 0}{3} = \mathbf{10.0}$$
   *(Output assumes missing data is a low value).*

#### Fix (Mathematical Consistency)
To maintain pure localized averaging, Keras defines `average_pool` as the average of *only* valid data present in the window. No synthetic data is injected into the computation.

$$\text{Window} = [10.0, 20.0, \text{Missing}]$$
$$\text{Valid Sum} = 10 + 20 = 30$$
$$\text{Valid Count} = 2$$
$$\text{Accurate Average} = \frac{30}{2} = \mathbf{15.0}$$

Reproduction script: https://colab.research.google.com/drive/1doBcueVhI88E6o2ociCWtj3JCww59fmp?usp=sharing

## Contributor Agreement

**Please review our [AI-Assisted Contribution Policy](../CONTRIBUTING.md#ai-assisted-contribution-policy) and check all boxes below before submitting your PR for review:**

- [x] I am a human, and not a bot.
- [x] I will be responsible for responding to review comments in a timely manner.
- [x] I will work with the maintainers to push this PR forward until submission.

**Note:** Failing to adhere to this agreement may result in your future PRs no longer being reviewed.

### CODE REVIEW

File: keras/src/ops/nn_test.py [hunk 1]
```diff
@@ -1772,12 +1772,53 @@ def test_average_pool_same_padding(self):
             ),
         )
 
+    def test_average_pool_same_padding_asymmetric(self):
+        data_format = backend.config.image_data_format()
```
Comment (hertschuh) [inline keras/src/ops/nn_test.py]: Instead of this, make this a parameterized test that verifies both "channels_first" and "channels_last".

File: keras/src/backend/torch/nn.py [hunk 2]
```diff
@@ -438,8 +441,18 @@ def average_pool(
             pool_size,
             strides,
             "channels_first",  # we're in channels_first here
-            "pooling",
+            "average_pooling",
```
Comment (hertschuh) [inline keras/src/backend/torch/nn.py]: While this "works", the usage of `_apply_same_padding` is really weird now.

The `operation_type` parameter is really treated as a boolean based on whether it is equal to `"pooling"` or not. But now the pooling operation doesn't even use `"pooling"`, it uses some arbitrary string that could be anything.

Looking at the code  for `_apply_same_padding`, the first 2 uses are actually useless because the default value of dilation_rate of `1` already handles this correctly.

So I would instead replace `operation_type` with a `padding_mode` argument with a default value of `"constant"` that you pass to `tnn.pad` at the end. Then you pass `"replicate"` in the convolution case of `_apply_same_padding` .

**Review by hertschuh — CHANGES_REQUESTED** [review]
Thanks for addressing this!

File: keras/src/ops/nn_test.py (hunk 1 above)
Comment (buildwithsuhana) [reply keras/src/ops/nn_test.py]: Done

File: keras/src/backend/torch/nn.py (hunk 2 above)
Comment (buildwithsuhana) [reply keras/src/backend/torch/nn.py]: Refactored _apply_same_padding to use a padding_mode argument (defaulting to constant) and simplified _compute_padding_length by removing the redundant if stride == 1 block.
Regarding the padding mode, I've updated `max_pool` to use `replicate` padding, and `conv` using `constant` padding (zero-pad), using replicate for convolution changes the expected edge results and causes failures. Convolution expects values outside the boundary to be zero so they add nothing to the sum, using replicate padding adds extra copies of edge pixels instead, which changes the final result.

**Review by hertschuh — APPROVED** [review]
Thanks!
